src/Mod/PartDesign/Gui/TaskExtrudeParameters.cpp

```diff
@@ -388,6 +390,94 @@ void TaskExtrudeParameters::addAxisToCombo(App::DocumentObject* linkObj, std::st
         lnk.setValue(linkObj, std::vector<std::string>(1, linkSubname));
 }
 
+void TaskExtrudeParameters::setCheckboxes(int index, bool isPocket)

```

Comment:
It may sound nitpicking but the boolean here is a code smell (called boolean blindness) because in places where the function is used you see this `setCheckboxes(index, true);` or this `setCheckboxes(index, false);` and for a reviewer it's not immediately clear what the intention is, hence boolean blindness. So he has to go to the class declaration to get its meaning.

A better way is to use an enum that can be declared as:
```
enum class Type {
  Pad,
  Pocket
};
```
Then in the code you will have to write `setCheckboxes(index, Type::Pocket);` or setCheckboxes(index, Type::Pad);`

Reply:
> It may sound nitpicking but the boolean here is a code smell (called boolean blindness) because in places where the function is used you see this `setCheckboxes(index, true);` or this `setCheckboxes(index, false);` and for a reviewer it's not immediately clear what the intention is, hence boolean blindness. 

I can change this to an enum. However I don't understand the term "blindness". I named the Boolean "isPocket", so it is clear what it is about. The IDEs like KDevelop, MSVC etc. show you thin info why typing. Also when you right. click on it, you get the info about the procedure definition. If I would have added a comment in the .h file, the IDE will also show me this. So with one click, the coders get the info.
When you don't have an IDE but use a text editor, then I agree an enums helps for readability.

I must admit I use bools because this is quicker ceded than adding an enum, since I write a procedure and in the IDE I can right-click on it to ad its definition in the header file. Therefore I only need to write once, the rest is automatically done by the IDE. (Only the position in in private/public etc. needs sometimes to be adapted.)

So should I in future prefer enums over bools?

Reply:
The enums are now in.

Reply:
> I can change this to an enum. However I don't understand the term "blindness". I named the Boolean "isPocket", so it is clear what it is about.

It only becomes clear if somebody who reads the switches to the class declaration. And it's only clear if the value is true, but what if it's false? Somebody who is not familiar with the code won't know that it will be a pad then.

To better understand the issue have a look at: https://doc.qt.io/archives/qq/qq13-apis.html#thebooleanparametertrap

> I must admit I use bools because this is quicker ceded than adding an enum, since I write a procedure and in the IDE I can right-click on it to ad its definition in the header file.

But you don't write the code only for you but for everybody else. And thus the client code should be as clear as possible so that there is no need to always jump between client code and class declaration.

Then with a boolean you only have two cases but what if in the future a new case will be added? Do you then add another boolean? An example of this bad practice is https://github.com/FreeCAD/FreeCAD/blob/f1d6c78cbe3c17a0b219e0f7b1443ec53463b80a/src/Mod/PartDesign/Gui/ReferenceSelection.h#L49 which in client code then e.g. looks like this https://github.com/FreeCAD/FreeCAD/blob/249c2c79705f3fa11c0f5c2d5d2c6fa0cfad4f3d/src/Mod/PartDesign/Gui/TaskDraftParameters.cpp#L215
Having six boolean parameters makes it really difficult to understand the code immediately.

Reply:
> To better understand the issue have a look at: https://doc.qt.io/archives/qq/qq13-apis.html#thebooleanparametertrap

Thanks, interesting article.

> An example of this bad practice is
> 
> https://github.com/FreeCAD/FreeCAD/blob/f1d6c78cbe3c17a0b219e0f7b1443ec53463b80a/src/Mod/PartDesign/Gui/ReferenceSelection.h#L50

I see.
I think the point is that everybody uses an IDE with proper information delivery. For example here with MSVC I just have too move the mouse over a 'true' and get its type and when I want also the whole procedure definition. But it is exactly this feature doesn't make you think about code readability for those without a proper IDE. I'll try to think about this more.

Reply:
An IDE doesn't help at all when reading the code on-line.